src/script/common/c_content.cpp

```diff
@@ -646,6 +660,36 @@ NodeBox read_nodebox(lua_State *L, int index)
 		if(lua_istable(L, -1))
 			nodebox.wall_side = read_aabb3f(L, -1, BS);
 		lua_pop(L, 1);
+
+		lua_getfield(L, index, "connect_up");
+		if (lua_istable(L, -1))
+			nodebox.connect_up = read_aabb3f_vector(L, -1, BS);
+		lua_pop(L, 1);
+
+		lua_getfield(L, index, "connect_down");
+		if (lua_istable(L, -1))
+			nodebox.connect_down = read_aabb3f_vector(L, -1, BS);
+		lua_pop(L, 1);
+
+		lua_getfield(L, index, "connect_n");
+		if (lua_istable(L, -1))
+			nodebox.connect_n = read_aabb3f_vector(L, -1, BS);
+		lua_pop(L, 1);
+
+		lua_getfield(L, index, "connect_e");
+		if (lua_istable(L, -1))
+			nodebox.connect_e = read_aabb3f_vector(L, -1, BS);
+		lua_pop(L, 1);
+
+		lua_getfield(L, index, "connect_s");
+		if (lua_istable(L, -1))
+			nodebox.connect_s = read_aabb3f_vector(L, -1, BS);
+		lua_pop(L, 1);
+
+		lua_getfield(L, index, "connect_w");
+		if (lua_istable(L, -1))
+			nodebox.connect_w = read_aabb3f_vector(L, -1, BS);
+		lua_pop(L, 1);
```

Comment:
I'm not convinced that this code is better if we'd (for instance) macroize this. Ultimately this is what the code does, and it's not doing hundreds of repetitions, plus you'd end up with 6 identical calls to a function.

Rolling these type of short loops up would mean all sorts of random macro's all over the code, or functions just for the sake of functions. My preference says to not make functions for a small amount of lines of code here.

These would also require a do {} while(0) macro as well. I can only find a few of those in the current code base, and I consider that a good thing.